src/core/qgsdistancearea.cpp

```diff
@@ -384,6 +400,12 @@ QgsPointXY QgsDistanceArea::computeSpheroidProject(
   double a = mSemiMajor;
   double b = mSemiMinor;
   double f = 1 / mInvFlattening;
+  if ( ( ( a < 0 ) && ( b < 0 ) ) ||
+       ( ( p1.x() < -180.0 ) || ( p1.x() > 180.0 ) || ( p1.y() < -85.05115 ) || ( p1.y() > 85.05115 ) ) )
```

Comment:
I will test this with the spatialite function that uses the rttopo function
* if no errors occur I will revert it to 90